packages/dds/task-manager/src/taskManager.ts

```diff
@@ -312,6 +312,34 @@ export class TaskManager extends SharedObject<ITaskManagerEvents> implements ITa
         return lockAcquireP;
     }
 
+    public subscribeToTask(taskId: string) {
+        const disconnectHandler = () => {
+            // Wait to be connected again and then re-submit volunteer op
+            this.connectionWatcher.once("connect", () => {
+                this.submitVolunteerOp(taskId);
+            });
+        };
+
+        const checkIfAbandoned = (eventTaskId: string) => {
+            if (eventTaskId !== taskId) {
+                return;
+            }
+
+            this.abandonWatcher.off("abandon", checkIfAbandoned);
+            this.connectionWatcher.off("disconnect", disconnectHandler);
+        };
+
+        this.abandonWatcher.on("abandon", checkIfAbandoned);
```

Comment:
Maybe it would be nice to represent state diagram of individual task and its relationship to queue. It looks like abandoning disconnected task thrown an error, unsure also what happens the my "volunteer" request in the queue on disconnect/connect etc. This is probably easiest to describe in some diagram.